subscription_integration_test.go

```diff
@@ -745,3 +745,136 @@ func TestSubscriptionAllFieldsWithTrialPeriodNeverExpires(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestSubscriptionModifications(t *testing.T) {
+	customer, err := testGateway.Customer().Create(&Customer{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	paymentMethod, err := testGateway.PaymentMethod().Create(&PaymentMethodRequest{
+		CustomerId:         customer.Id,
+		PaymentMethodNonce: FakeNonceTransactable,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log(customer)
+
+	g := testGateway.Subscription()
+
+	// Create
+	sub, err := g.Create(&SubscriptionRequest{
+		PaymentMethodToken: paymentMethod.GetToken(),
+		PlanId:             "test_plan_2",
+	})
+
+	t.Log("sub1", sub)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	if sub.Id == "" {
+		t.Fatal("invalid subscription id")
+	}
+
+	// Add AddOn
+	sub2, err := g.Update(&SubscriptionRequest{
+		Id: sub.Id,
+		AddOns: &ModificationsRequest{
+			Add: []AddModificationRequest{
+				{
+					InheritedFromID: "test_add_on",
+					ModificationRequest: ModificationRequest{
+						Amount:       NewDecimal(300, 2),
+						Quantity:     1,
+						NeverExpires: true,
+					},
+				},
+			},
+		},
+		Discounts: &ModificationsRequest{
+			Add: []AddModificationRequest{
+				{
+					InheritedFromID: "test_discount",
+					ModificationRequest: ModificationRequest{
+						Amount:       NewDecimal(100, 2),
+						Quantity:     1,
+						NeverExpires: true,
+					},
+				},
+			},
+		},
+	})
+
+	t.Log("sub2", sub2)
```

Comment:
Not really, I based it off the pattern used in existing tests in this file and in others so they were uniform. We should probably cleanup the logging so it's more meaningful. I'll leave that for another time.